servicetalk-http-api/src/main/java/io/servicetalk/http/api/StreamingContentCodec.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.concurrent.api.Publisher;
+
+/**
+ * API for HTTP <a href="https://tools.ietf.org/html/rfc7231#section-3.1.2.1">Content Codings</a>.
+ *
+ * Used to encode and decode {@link Publisher} of {@link Buffer} parts.
+ * This instance is shared therefore it must provide thread safety semantics.
+ */
+public interface StreamingContentCodec extends ContentCodec {
+
+    /**
+     * Take a {@link Publisher} of {@link Buffer} and encode its contents resulting in a
+     * {@link Publisher} of {@link Buffer} with the encoded contents.
+     *
+     * @param from the {@link Publisher} buffer to encode
+     * @param allocator the {@link BufferAllocator} to use for allocating auxiliary buffers or the returned buffer
+     * @return {@link Publisher} the result publisher with the buffers encoded
+     */
+    default Publisher<Buffer> encode(Publisher<Buffer> from, BufferAllocator allocator) {
+        return from.map((buffer -> encode(buffer, allocator)));
```

Comment:
I do not see much value in having an extended interface that adds only default methods. In this case, we can just add default methods to `ContentCodec` interface.

Another approach will be to have 2 independent interfaces: one for streamin, one for non-streaming use-case. They will have a common `CharSequence name()` method, which is fine to duplicate.

Each approach has pros and cons:
- With a single interface:
-- users who care only about a streaming variant will also need to implement non-streaming variant. They probably can just thrown `UnsupportedOperationException`.
-- users who use non-streaming implementation for streaming case may be broken (depends on compression algorithm). I do not expect it to happen with our implementations.

- With 2 interfaces builders will need to have 2  methods: for streaming (`buildStreaming()`) and non-streaming (`build()`) variant.

I'm fine with both approaches. If we anticipate a protocol in the future that needs both variants, having everything in a single interface will be better.

@Scottmitch wdyt?